### PR TITLE
[WIP] Cache between cluster state changes, allow checking of specific settings

### PR DESCRIPTION
**Pull request by danhermann**
These two changes allow the `ConsistentSettingsService` to be used with the new hash processor:

1. The consistency of a setting relative to the current master node can change only when the cluster's master node changes, so the consistency of settings can be cached and re-computed only when the cluster's master node changes. The hash processor must ensure that its hash key is consistent on every execution and checking this cached value will be faster.
1. The hash processor should fail only if its own hash key is inconsistent, not if other unrelated settings are inconsistent. The overloaded `areAllConsistent(Collection<Setting<?>>)` method allows the hash processor to query the consistency of only the settings that are relevant to it.

### CODE REVIEW

**Comment by danhermann** [comment]
@elastic/es-core-infra, I'd appreciate any input on this approach to adapting `ConsistentSettingsService` for use with the hash processor. If this approach is acceptable, I can fix the one remaining IT that is failing.

File: server/src/main/java/org/elasticsearch/common/settings/ConsistentSettingsService.java [hunk 1]
```diff
@@ -44,22 +49,19 @@
...
  */
-public final class ConsistentSettingsService {
+public final class ConsistentSettingsService implements ClusterStateListener {
     private static final Logger logger = LogManager.getLogger(ConsistentSettingsService.class);
 
     private final Settings settings;
     private final ClusterService clusterService;
     private final Collection<Setting<?>> secureSettingsCollection;
     private final SecretKeyFactory pbkdf2KeyFactory;
+    private volatile Map<String, Boolean> consistencyCache;
```
Comment (jakelandis) [inline server/src/main/java/org/elasticsearch/common/settings/ConsistentSettingsService.java]: Do we need this cache, and do we need to populate it on cluster state change ?  I see it the cluster state work is protected by a change in master node ids, but it seems that a simpler solution is check the consistency every time with every call to `areAllConsistent`

**Review by jakelandis — CHANGES_REQUESTED** [review]
A couple comments .. the larger question is do we need the cache and cluster state listener ?

**Comment by danhermann** [comment]
Closing as these changes will not be necessary for the hash processor.

**Pull request closed without merging**